core/controller/src/main/scala/whisk/core/controller/GatewayRoutesAPI.scala

```diff
@@ -0,0 +1,191 @@
+import spray.http.StatusCodes._
+import whisk.core.entity._
+import scala.util.Success
+import whisk.core.entity.EntityPath
+import whisk.core.entity.Identity
+import scala.concurrent._
+import scala.concurrent.{ Future , Promise }
+import scala.concurrent.duration._
+import scala.util.{Failure,Success}
+import whisk.core.entity.WhiskAuth
+import whisk.core.entity.Subject
+import spray.http._
+import spray.client.pipelining._
+import akka.actor.ActorSystem
+import scala.concurrent.Future
+import spray.http.BasicHttpCredentials
+import scala.concurrent.ExecutionContext.Implicits.global
+import spray.http.Uri
+import spray.http.Uri.Path
+import whisk.common.TransactionId
+import whisk.core.WhiskConfig
+import whisk.core.entity._
+import whisk.core.entity.{WhiskAuthStore}
+import akka.event.Logging.LogLevel
+
+class gatewayRoutesApi(
+        val apipath: String,
+        val apiversion: String,
+        val verbosity: LogLevel,
+        val config: WhiskConfig,
+        implicit val actorSystem: ActorSystem){
+        protected implicit val authStore = WhiskAuthStore.datastore(config)
+        private val url = Uri(s"http://localhost:${config.servicePort}")
+        private val key = Await.result(WhiskAuth.get(authStore, Subject("whisk.system"),false)(TransactionId.unknown), 5.seconds)
+        private val validCredentials = BasicHttpCredentials(key.authkey.uuid(), key.authkey.key())
+        /**
+        * Handles invoking create action
+        *
+        * @return Future[String] from action result
+        */
+        def invokeActionCreate(requestBody: String ) : Future[String]  = {
+            val actionPath = Path("/api/"+apiversion) / "namespaces" / "whisk.system" / "actions" / "routemgmt" / "createRoute"
+            val pipeline: HttpRequest => Future[HttpResponse] = (
+                addCredentials(validCredentials)
+                   ~> sendReceive
+            )
+            val p = Promise[String]()
+            val response: Future[HttpResponse] = pipeline(Post(url.withPath(actionPath).toString.concat("?blocking=true"), HttpEntity(ContentTypes.`application/json`, requestBody))  )
+            response.onComplete({
+                case Success(result: HttpResponse) => {
+                  p trySuccess result.entity.asString
+                }
+                case Failure(error) => {
+                  p tryFailure error
+                }
+             })
+             p.future
+       }
+       /**
+        ** Handles invoking get actions
+        **
+        * @return Future[String] from action result
+        */
+        def invokeActionGet(requestBody: String ) : Future[String]  = {
+            val actionPath = Path("/api/"+apiversion) / "namespaces" / "whisk.system" / "actions" / "routemgmt" / "getApi"
+            val pipeline: HttpRequest => Future[HttpResponse] = (
+                addCredentials(validCredentials)
+                    ~> sendReceive
+            )
+            val p = Promise[String]()
```

Comment:
I suggest doing this change sooner rather than later so that you can avoid accidentally leaking a promise or forgetting to complete it as the code evolves.